generators/chipyard/src/main/scala/TestHarness.scala

```diff
@@ -19,32 +23,96 @@ trait HasTestHarnessFunctions {
 }
 
 trait HasHarnessSignalReferences {
+  // clock/reset of the chiptop reference clock (can be different than the implicit harness clock/reset)
   def harnessClock: Clock
   def harnessReset: Reset
   def dutReset: Reset
   def success: Bool
 }
 
+class HarnessClockInstantiator {
+  private var _clockMap: HashMap[String, (Double, ClockBundle)] = HashMap.empty
+
+  // request a clock bundle at a particular frequency
+  def getClockBundle(name: String, freqRequested: Double): ClockBundle = {
```

Comment:
I think this should be named `requestClockBundle` as it looks much more like a setter than a getter which the `getClockBundle` name suggested to me at first. I had to read the comment to understand but if you change the name the comment basically isn't needed as its now obvious from the name.

Reply:
I think `Double` is fine. However, the DTS frequency associated with the buses uses a `BigInt` (https://github.com/chipsalliance/rocket-chip/blob/ca2567ebc200e7027cbbe8c2d3e70378c6bab445/src/main/scala/tilelink/BusWrapper.scala#L26-L27). IIRC there are other places that use `Int` and `BigInt` as well... I can try going through and enforcing the `Double` though.